pywinauto/__init__.py

```diff
@@ -104,24 +104,39 @@ def _get_com_threading_mode(module_sys):
 
 
     class Desktop(object):
-
         """Simple class to call something like ``Desktop().WindowName.ControlName.method()``"""
 
-        def __init__(self, backend=None):
+        def __init__(self, backend="win32"):
             """Create desktop element description"""
-            if backend:
-                self.backend = backend
-            else:
-                self.backend = backends.registry.name
+            if backend not in backends.registry.backends:
+                raise ValueError('Backend "{0}" is not registered!'.format(backend))
+            self.backend = backends.registry.backends[backend]
 
         def window(self, **criterion):
             """Create WindowSpecification object for top-level window"""
             if 'top_level_only' not in criterion:
                 criterion['top_level_only'] = True
             if 'backend' not in criterion:
-                criterion['backend'] = self.backend
+                criterion['backend'] = self.backend.name
             return WindowSpecification(criterion)
 
+        def windows(self, **criterion):
+            """Return a list of wrapped top level windows"""
+
+            if 'backend' in criterion:
+                raise ValueError('Do not override backend!')
+
+            if 'visible_only' not in criterion:

```

Comment:
Maybe conditions could be covered by auto tests better. I mean covering exception and `visible_only`/`enabled_only` in `**kwargs`.

Reply:
I think I undestand how can i do this check.